contracts/BytesLib.sol

```diff
@@ -426,6 +426,27 @@ library BytesLib {
                         cb := 0
                     }
                 }
+
+                // Only if still successful
+                // For <1 word tail bytes
+                if gt(success, 0) {
+                    let numTailBytes := sub(add(endWord, 0x20), mc)
```

Comment:
I don't believe this is correct. How are we calculating the number of tailing bytes if we're only using memory pointers and never the length?

Can you verify this for me, please, @djb15 ? Did you create any tests for this to try it out?

Reply:
`endWord` is initially set to the length of the input bytes + the initial pointer (minus 0x20 to save gas). So when we subtract mc (which is the current pointer location) we should have the number of bytes remaining.

Haven't added any tests here but have verified this against an internal failing test. Happy to spin this out into it's own method to minimise risk + not increase gas costs for normal use cases

Reply:
But `mc` is always a multiple of 32, no? Which makes the resulting `numTailBytes` either zero or thirty two. Just from intuition I'd say I'm pretty sure `length` needs to come into play at some point because all other cars are always multiple of 32.

Reply:
Correct, `mc` is always a multiple of 32, but `endWord` might not be. 

For example, let's say `_preBytes` byte array is 8 bytes long and the start pointer is 0x00. So `endWord = 0x08`. Initially `mc = 0x20` so the first (existing) loop won't be executed.

We get to the changes that I've made: `numTailBytes = (0x20 + 0x08) - 0x20 = 0x08` which is the correct number of tail bytes

Reply:
Lol, shit, ok, you're right! Im responding on my phone and when skimming it got mislead by the naming of the variable. `endWord` is a derivative of `length`. 

To be honest I don't especially like the lack of syntactic meaning of the loop condition. Let me take another look when I get home.

Reply:
OK, back home and finally reasoning properly about this. Here is my proposal to finish the PR:

* change `endWord` var name to `endMinusWord`.
* change this line to: `let numTailBytes := and(length, 0x1F)`

The first change, the var name, just because it is more factually representative of the value it holds.
The second change, of the number of trailing bytes, because not only does it improve on gas, but it greatly improves on syntactic meaning and readability. Since what we want to get is the trailing number of bytes, let's just divide the length by 32 and keep the remainder (that's what ANDing with `32 - 1 = 0x1F` does).

Do you agree?

Reply:
Nice! Yeah I agree with both of these changes, thanks for the suggestions

Reply:
These have been resolved as discussed